controllers/ptpconfig_controller.go

```diff
@@ -112,7 +127,325 @@ func (r *PtpConfigReconciler) syncPtpConfig(ctx context.Context, ptpConfigList *
+		h.Write([]byte(";"))
+	}
+
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+// extractSaFileFromPtp4lConf scans the [global] section for a line starting with 'sa_file'
+// and returns the remainder of the line as the path.
+func extractSaFileFromPtp4lConf(conf string) (string, bool) {
+	inGlobal := false
+	sectionHeader := regexp.MustCompile(`^\s*\[[^\]]+\]\s*$`)
+	for _, raw := range strings.Split(conf, "\n") {
+		line := raw
+		// strip inline comments '#' and ';'
+		if idx := strings.IndexAny(line, "#;"); idx >= 0 {
+			line = line[:idx]
+		}
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		if sectionHeader.MatchString(line) {
+			if strings.EqualFold(strings.TrimSpace(line), "[global]") {
+				inGlobal = true
+				continue
+			}
+			if inGlobal {
+				// leaving global section
+				break
+			}
+			continue
+		}
+		if !inGlobal {
+			continue
+		}
+		// check for sa_file directive
+		if strings.HasPrefix(line, "sa_file") {
+			parts := strings.Fields(line)
+			if len(parts) >= 2 {
+				return parts[1], true
+			}
+		}
+	}
+	return "", false
+}
+
+// injectPtpSecurityVolume adds a secret volume and mounts to the DaemonSet
+func injectPtpSecurityVolume(ds *appsv1.DaemonSet, secretName string, secretKey string, secretKeys map[string]struct{}, saFilePaths []string, secretHash string) {
+	// 1. Clean up any existing ptp-security-volume
+	var cleanedVolumes []corev1.Volume
+	for _, vol := range ds.Spec.Template.Spec.Volumes {
+		if vol.Name != "ptp-security-volume" {
+			cleanedVolumes = append(cleanedVolumes, vol)
+		}
+	}
+	ds.Spec.Template.Spec.Volumes = cleanedVolumes
+
+	// 2. Add the new Volume
+	ds.Spec.Template.Spec.Volumes = append(ds.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: "ptp-security-volume",
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	})
+
+	// 3. Add secret hash annotation to trigger pod restart on secret change
+	if ds.Spec.Template.Annotations == nil {
+		ds.Spec.Template.Annotations = make(map[string]string)
+	}
+	annotationKey := "ptp.openshift.io/secret-hash"
+	oldHash := ds.Spec.Template.Annotations[annotationKey]
+	if oldHash != secretHash {
+		glog.Infof("Secret hash changed (old: %s, new: %s) - pods will be restarted", oldHash, secretHash)
+		ds.Spec.Template.Annotations[annotationKey] = secretHash
+	} else {
+		glog.Infof("Secret hash unchanged (%s) - no pod restart needed", secretHash)
+	}
+
+	// 4. Find the target container
+	var targetContainer *corev1.Container
+	for i := range ds.Spec.Template.Spec.Containers {
+		if ds.Spec.Template.Spec.Containers[i].Name == "linuxptp-daemon-container" {
+			targetContainer = &ds.Spec.Template.Spec.Containers[i]
+			break
+		}
+	}
+
+	if targetContainer == nil {
+		glog.Errorf("Could not find container 'linuxptp-daemon-container' to inject VolumeMount")
+		return
+	}
+
+	// 5. Clean up any existing ptp-security-volume mounts
+	var cleanedMounts []corev1.VolumeMount
```

Comment:
The volume name cannot be hardcoded(same as above use the file name in the sa_file). I would let this function only insert a new volume. At every re-computation of the volume mounts, both for security volumes and base ptp-operator volumes, the template should be cleaned and all volumes re-added, this way there is no need to known the individual volume names to delete.